xscen/scripting.py

```diff
@@ -338,3 +344,119 @@ def skippable(seconds: int = 2, task: str = "", logger: logging.Logger = None):
         err("Received SIGINT. Do it again to stop the script.")
         time.sleep(seconds)
         inf(f"Skipping {task}.")
+
+
+def save_and_update(
+    ds: xr.Dataset,
+    pcat: ProjectCatalog,
+    path: Union[str, os.PathLike] = None,
+    file_format: str = None,
+    build_path_kwargs: dict = None,
+    save_kwargs: dict = None,
+    update_kwargs: dict = None,
+):
+    """
+    Construct the path, save and delete.
+
+    This function can be used after each task of a workflow.
+
+    Parameters
+    ----------
+    ds: xr.Dataset
+        Dataset to save.
+    pcat: ProjectCatalog
+        Catalog to update after saving the dataset.
+    path: str | os.pathlike
+        Path where to save the dataset.
+        If the string contains variables in curly bracket. They will be filled by catalog attributes.
+        If None, the `catutils.build_path` fonction will be used to create a path.
+    file_format: {'nc', 'zarr'}
+        Format of the file.
+        If None, look for the following in order: build_path_kwargs['format'], a suffix in path, ds.attrs['cat:format'].
+        If nothing is found, it will default to zarr.
+    build_path_kwargs: dict
+        Arguments to pass to `build_path`.
+    save_kwargs:
+        Arguments to pass to `save_to_netcdf` or `save_to_zarr`.
+    update_kwargs: dict
+        Arguments to pass to `update_from_ds`.
+    """
+    build_path_kwargs = build_path_kwargs or {}
+    save_kwargs = save_kwargs or {}
+    update_kwargs = update_kwargs or {}
+
+    # try to guess file format if not given.
+    if file_format is None:
+        file_format = build_path_kwargs.get("format", None)
+        if path is not None and "." in path:
+            file_format = Path(path).suffix.split(".")[-1]
+        else:
+            file_format = ds.attrs.get("cat:format", "zarr")
+
+    # get path
+    if path is not None:
+        path = path.format(**get_cat_attrs(ds))  # fill path with attrs
```

Comment:
```suggestion
        path = str(path).format(**get_cat_attrs(ds))  # fill path with attrs
```
Since we accept Path objects too.